src/components/oba/TripDetailsPane.svelte

```diff
@@ -0,0 +1,141 @@
+<script>
+	import { onMount, onDestroy } from 'svelte';
+	import { faBus, faLocationDot } from '@fortawesome/free-solid-svg-icons';
+	import { faSquare } from '@fortawesome/free-regular-svg-icons';
+	import { FontAwesomeIcon } from '@fortawesome/svelte-fontawesome';
+
+	export let tripId;
+	export let serviceDate = null;
+
+	let tripDetails = null;
+	let routeInfo = null;
+	let stopInfo = {};
+	let error = null;
+	let interval;
+	let currentStopIndex = -1;
+	let busPosition = 0;
+
+	function formatTime(seconds) {
+		if (!seconds) return '';
+		const date = new Date(seconds * 1000);
+		return date.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });
+	}
+
+	function calculateBusPosition() {
+		if (tripDetails && tripDetails.status && tripDetails.status.position) {
+			const { lat, lon } = tripDetails.status.position;
+
+			busPosition = tripDetails.schedule.stopTimes.findIndex((stop, index, array) => {
+				const nextStop = array[index + 1];
+				if (!nextStop) return true;
+				const stopLat = stopInfo[stop.stopId].lat;
+				const stopLon = stopInfo[stop.stopId].lon;
+				const nextStopLat = stopInfo[nextStop.stopId].lat;
+				const nextStopLon = stopInfo[nextStop.stopId].lon;
+				return (lat >= stopLat && lat < nextStopLat) || (lon >= stopLon && lon < nextStopLon);
+			});
+		}
+	}
+
+	async function loadTripDetails() {
+		try {
+			let url = `/api/oba/trip-details/${tripId}?includeTrip=true&includeSchedule=true&includeStatus=true`;
+			if (serviceDate) {
+				url += `&serviceDate=${serviceDate}`;
+			}
+			const response = await fetch(url);
+			if (response.ok) {
+				const data = await response.json();
+				tripDetails = data.data.entry;
+
+				if (data.data.references && data.data.references.routes) {
```

Comment:
now that you have `data` extracted above, you can use that along with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to simplify these conditions:

```
if (data?.references?.routes) {
  // ...
}
```